From 8c8522f1c5c8362ec683b5464e99a95bc735677f Mon Sep 17 00:00:00 2001
From: Eric Blake <eblake@redhat.com>
Date: Tue, 17 Jul 2012 12:02:12 -0600
Subject: [PATCH] doc: fix texinfo location reports

Otherwise, newer texinfo parses this line as a line directive, and
any error later in the file will claim to be from "conftest.c" instead
of "autoconf.texi".

* doc/autoconf.texi (Generating Sources): Avoid confusing newer
texinfo into thinking we had a line directive.
Reported by Stefano Lattarini, fix suggested by Patrice Dumas.
---
 doc/autoconf.texi |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/doc/autoconf.texi b/doc/autoconf.texi
index 75a204f..78a2c67 100644
--- a/doc/autoconf.texi
+++ b/doc/autoconf.texi
@@ -9155,7 +9155,7 @@ on a system with @command{gcc} installed, results in:
 @example
 @c If you change this example, adjust tests/compile.at:AC_LANG_SOURCE example.
 @dots{}
-# 1 "conftest.c"
+@asis{#} 1 "conftest.c"
 
 #define PACKAGE_NAME "Hello"
 #define PACKAGE_TARNAME "hello"
@@ -9200,7 +9200,7 @@ on a system with @command{gcc} installed, results in:
 @example
 @c If you change this example, adjust tests/compile.at:AC_LANG_PROGRAM example.
 @dots{}
-# 1 "conftest.c"
+@asis{#} 1 "conftest.c"
 
 #define PACKAGE_NAME "Hello"
 #define PACKAGE_TARNAME "hello"
-- 
1.7.2.5
From a357718b081f1678748ead5d7cb67c766c930441 Mon Sep 17 00:00:00 2001
From: Patrice Dumas <pertusus@free.fr>
Date: Wed, 18 Jul 2012 13:41:45 -0600
Subject: [PATCH] doc: fix texinfo macro usage

The texinfo manual recommends avoiding the use of a trailing @c in
any macro designed to be used inline (as is the case with our ovar
and dvar macros).  Furthermore, passing '@\n' in the middle of a
macro call is much different than passing '@\n' between arguments
of a @defmac for line continuation.

* doc/autoconf.texi (ovar, dvar): Don't end macro with @c, since
these macros are designed to be embedded in one-line usage.
(Fortran Compiler): Don't split @dvar.
* THANKS: Update.
Reported by Stefano Lattarini.

Signed-off-by: Eric Blake <eblake@redhat.com>
Copyright-paperwork-exempt: Yes
---
 THANKS            |    1 +
 doc/autoconf.texi |   24 ++++++++++++------------
 2 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/THANKS b/THANKS
index 25734d5..8682c87 100644
--- a/THANKS
+++ b/THANKS
@@ -316,6 +316,7 @@ Olly Betts                  olly@survex.com
 Ossama Othman               ossama@debian.org
 Pallav Gupta                pallavgupta@gmail.com
 Paolo Bonzini               bonzini@gnu.org
+Patrice Dumas               pertusus@free.fr
 Patrick Tullmann            tullmann@cs.utah.edu
 Patrick Welche              prlw1@newn.cam.ac.uk
 Paul Berrevoets             paul@swi.com
diff --git a/doc/autoconf.texi b/doc/autoconf.texi
index 78a2c67..506d966 100644
--- a/doc/autoconf.texi
+++ b/doc/autoconf.texi
@@ -15,7 +15,7 @@
 @c The ARG is an optional argument.  To be used for macro arguments in
 @c their documentation (@defmac).
 @macro ovar{varname}
-@r{[}@var{\varname\}@r{]}@c
+@r{[}@var{\varname\}@r{]}
 @end macro
 
 @c @dvar(ARG, DEFAULT)
@@ -23,7 +23,7 @@
 @c The ARG is an optional argument, defaulting to DEFAULT.  To be used
 @c for macro arguments in their documentation (@defmac).
 @macro dvar{varname, default}
-@r{[}@var{\varname\} = @samp{\default\}@r{]}@c
+@r{[}@var{\varname\} = @samp{\default\}@r{]}
 @end macro
 
 @c Handling the indexes with Texinfo yields several different problems.
@@ -8014,10 +8014,10 @@ variables, respectively.  The computed linker flags are cached in
 @code{ac_cv_f77_libs} or @code{ac_cv_fc_libs}, respectively.
 @end defmac
 
-@defmac AC_F77_DUMMY_MAIN (@ovar{action-if-found}, @dvar{action-if-not-found, @
-  AC_MSG_FAILURE})
-@defmacx AC_FC_DUMMY_MAIN (@ovar{action-if-found}, @dvar{action-if-not-found, @
-  AC_MSG_FAILURE})
+@defmac AC_F77_DUMMY_MAIN (@ovar{action-if-found}, @
+  @dvar{action-if-not-found, AC_MSG_FAILURE})
+@defmacx AC_FC_DUMMY_MAIN (@ovar{action-if-found}, @
+  @dvar{action-if-not-found, AC_MSG_FAILURE})
 @acindex{F77_DUMMY_MAIN}
 @cvindex F77_DUMMY_MAIN
 @acindex{FC_DUMMY_MAIN}
@@ -8268,8 +8268,8 @@ results in @code{ac_cv_fc_srcext_@var{ext}} and
 @code{ac_cv_fc_pp_srcext_@var{ext}} variables, respectively.
 @end defmac
 
-@defmac AC_FC_PP_DEFINE (@ovar{action-if-success}, @dvar{action-if-failure, @
-  AC_MSG_FAILURE})
+@defmac AC_FC_PP_DEFINE (@ovar{action-if-success}, @
+  @dvar{action-if-failure, AC_MSG_FAILURE})
 @acindex{FC_PP_DEFINE}
 @caindex fc_pp_define
 
@@ -8287,8 +8287,8 @@ The result of this test is cached in the @code{ac_cv_fc_pp_define}
 variable.
 @end defmac
 
-@defmac AC_FC_FREEFORM (@ovar{action-if-success}, @dvar{action-if-failure, @
-  AC_MSG_FAILURE})
+@defmac AC_FC_FREEFORM (@ovar{action-if-success}, @
+  @dvar{action-if-failure, AC_MSG_FAILURE})
 @acindex{FC_FREEFORM}
 @caindex fc_freeform
 
@@ -8314,8 +8314,8 @@ The result of this test, or @samp{none} or @samp{unknown}, is cached in
 the @code{ac_cv_fc_freeform} variable.
 @end defmac
 
-@defmac AC_FC_FIXEDFORM (@ovar{action-if-success}, @dvar{action-if-failure, @
-  AC_MSG_FAILURE})
+@defmac AC_FC_FIXEDFORM (@ovar{action-if-success}, @
+  @dvar{action-if-failure, AC_MSG_FAILURE})
 @acindex{FC_FIXEDFORM}
 @caindex fc_fixedform
 
-- 
1.7.2.5
From 0bd5dbd9b271ade96d2a7783e465b79ec194e8af Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Tue, 13 Sep 2016 17:51:18 -0700
Subject: doc: port to Texinfo 6.3

* doc/autoconf.texi: Remove obsolete @setcontentsaftertitlepage
that provokes a warning from Texinfo 6.3.
---
 doc/autoconf.texi | 3 ---
 1 file changed, 3 deletions(-)

(limited to 'doc/autoconf.texi')

diff --git a/doc/autoconf.texi b/doc/autoconf.texi
index 029ddd65..ddbeddfb 100644
--- a/doc/autoconf.texi
+++ b/doc/autoconf.texi
@@ -6,9 +6,6 @@
 @settitle Autoconf
 @documentencoding UTF-8
 @setchapternewpage odd
-@ifnothtml
-@setcontentsaftertitlepage
-@end ifnothtml
 @finalout
 
 @c @ovar(ARG)
-- 
cgit v1.2.1
From 565a6dc50cfa01cec2fb4db894026689cdf4970c Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Wed, 14 Sep 2016 12:14:45 -0700
Subject: doc: resplit lines to avoid Texinfo chaos

* doc/autoconf.texi (Introduction): Resplit lines to avoid
Texinfo 6.3 incompatibility with Perl 5.22.  See:
http://lists.gnu.org/archive/html/bug-texinfo/2016-09/msg00037.html
---
 doc/autoconf.texi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

(limited to 'doc/autoconf.texi')

diff --git a/doc/autoconf.texi b/doc/autoconf.texi
index d3ccf197..9ad7dc1c 100644
--- a/doc/autoconf.texi
+++ b/doc/autoconf.texi
@@ -693,8 +693,8 @@ nature of God.  ``Surely a Physicist,'' said the physicist, ``because
 early in the Creation, God made Light; and you know, Maxwell's
 equations, the dual nature of electromagnetic waves, the relativistic
 consequences@enddots{}'' ``An Engineer!,'' said the engineer, ``because
-before making Light, God split the Chaos into Land and Water; it takes a
-hell of an engineer to handle that big amount of mud, and orderly
+before making Light, God split the Chaos into Land and Water; it takes
+a hell of an engineer to handle that big amount of mud, and orderly
 separation of solids from liquids@enddots{}'' The computer scientist
 shouted: ``And the Chaos, where do you think it was coming from, hmm?''
 
-- 
cgit v1.2.1
